### ISSUE TITLE
Race detected during TestRegisterCandidate

### ISSUE THREAD

**Issue by AnnaShaleva** [nspcc-dev/neo-go#2868]
It's likely to be the test's issue, but we'd better ensure that everything is OK with native cache handling.
```
2023-01-11T09:13:05.0472030Z === CONT  TestRegisterCandidate
2023-01-11T09:13:05.0472970Z     logger.go:130: 2023-01-11T09:08:00.607Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0473940Z     logger.go:130: 2023-01-11T09:08:00.608Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2023-01-11T09:13:05.0475050Z     logger.go:130: 2023-01-11T09:08:00.619Z	DEBUG	done processing headers	{"headerIndex": 4, "blockHeight": 3, "took": "263.117µs"}
2023-01-11T09:13:05.0476470Z     logger.go:130: 2023-01-11T09:08:00.620Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0477840Z     logger.go:130: 2023-01-11T09:08:00.621Z	DEBUG	processing rpc request	{"method": "invokescript", "params": "[DCECs2Ir9AF73+MXxYrtX0x1PyBrfbiWBG+n13S7xL9/jcIMFO6eoiwn40vQFI/EEI4I906PUEiyEsAfDAR2b3RlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtSOQ== ]"}
2023-01-11T09:13:05.0479300Z     logger.go:130: 2023-01-11T09:08:00.623Z	DEBUG	processing rpc request	{"method": "getblockcount", "params": "[]"}
2023-01-11T09:13:05.0481590Z     logger.go:130: 2023-01-11T09:08:00.624Z	DEBUG	processing rpc request	{"method": "calculatenetworkfee", "params": "[AI6bIKo49C0AAAAAAAAAAAAAAAAABwAAAAHunqIsJ+NL0BSPxBCOCPdOj1BIsgEAXgwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CDBTunqIsJ+NL0BSPxBCOCPdOj1BIshLAHwwEdm90ZQwU9WPqQLwoPU0OBcSOowWz8qBzQO9BYn1bUjkBACgMIQKzYiv0AXvf4xfFiu1fTHU/IGt9uJYEb6fXdLvEv3+NwkFW57Mn]"}
2023-01-11T09:13:05.0484830Z     logger.go:130: 2023-01-11T09:08:00.625Z	DEBUG	processing rpc request	{"method": "sendrawtransaction", "params": "[AI6bIKo49C0AAAAAAEDaEgAAAAAABwAAAAHunqIsJ+NL0BSPxBCOCPdOj1BIsgEAXgwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CDBTunqIsJ+NL0BSPxBCOCPdOj1BIshLAHwwEdm90ZQwU9WPqQLwoPU0OBcSOowWz8qBzQO9BYn1bUjkBQgxAiredPXvdCA4tj9Gqdu5PpBynSY9atPuF9POuurylBwTQFjAw7eoFQroZwtx1o5dKaBcn73H43pPQggEBAP2AlSgMIQKzYiv0AXvf4xfFiu1fTHU/IGt9uJYEb6fXdLvEv3+NwkFW57Mn]"}
2023-01-11T09:13:05.0486860Z     logger.go:130: 2023-01-11T09:08:00.722Z	DEBUG	done processing headers	{"headerIndex": 5, "blockHeight": 4, "took": "229.53µs"}
2023-01-11T09:13:05.0487730Z     logger.go:130: 2023-01-11T09:08:00.728Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0488670Z     logger.go:130: 2023-01-11T09:08:00.729Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2023-01-11T09:13:05.0489620Z     logger.go:130: 2023-01-11T09:08:00.739Z	DEBUG	processing rpc request	{"method": "getcommittee", "params": "[]"}
2023-01-11T09:13:05.0490540Z     logger.go:130: 2023-01-11T09:08:00.741Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0491440Z     logger.go:130: 2023-01-11T09:08:00.742Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2023-01-11T09:13:05.0492580Z     logger.go:130: 2023-01-11T09:08:00.754Z	DEBUG	processing rpc request	{"method": "getcandidates", "params": "[]"}
2023-01-11T09:13:05.0493680Z     logger.go:130: 2023-01-11T09:08:00.755Z	DEBUG	processing rpc request	{"method": "getcommittee", "params": "[]"}
2023-01-11T09:13:05.0494570Z     logger.go:130: 2023-01-11T09:08:00.757Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0495460Z     logger.go:130: 2023-01-11T09:08:00.758Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2023-01-11T09:13:05.0496740Z     logger.go:130: 2023-01-11T09:08:00.768Z	DEBUG	processing rpc request	{"method": "invokefunction", "params": "[ef4073a0f2b305a38ec4050e4d3d28bc40ea63f5 getAccountState ]"}
2023-01-11T09:13:05.0498220Z     logger.go:130: 2023-01-11T09:08:00.770Z	DEBUG	processing rpc request	{"method": "invokefunction", "params": "[ef4073a0f2b305a38ec4050e4d3d28bc40ea63f5 decimals ]"}
2023-01-11T09:13:05.0499810Z     logger.go:130: 2023-01-11T09:08:00.773Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0500750Z     logger.go:130: 2023-01-11T09:08:00.774Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2023-01-11T09:13:05.0501590Z     logger.go:130: 2023-01-11T09:08:00.786Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0502890Z     logger.go:130: 2023-01-11T09:08:00.787Z	DEBUG	processing rpc request	{"method": "invokescript", "params": "[CwwU7p6iLCfjS9AUj8QQjgj3To9QSLISwB8MBHZvdGUMFPVj6kC8KD1NDgXEjqMFs/Kgc0DvQWJ9W1I5 ]"}
2023-01-11T09:13:05.0503910Z     logger.go:130: 2023-01-11T09:08:00.789Z	DEBUG	processing rpc request	{"method": "getblockcount", "params": "[]"}
2023-01-11T09:13:05.0505600Z     logger.go:130: 2023-01-11T09:08:00.790Z	DEBUG	processing rpc request	{"method": "calculatenetworkfee", "params": "[AHYJDnBm8y0AAAAAAAAAAAAAAAAACAAAAAHunqIsJ+NL0BSPxBCOCPdOj1BIsgEAPAsMFO6eoiwn40vQFI/EEI4I906PUEiyEsAfDAR2b3RlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtSOQEAKAwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CQVbnsyc=]"}
2023-01-11T09:13:05.0508230Z     logger.go:130: 2023-01-11T09:08:00.791Z	DEBUG	processing rpc request	{"method": "sendrawtransaction", "params": "[AHYJDnBm8y0AAAAAAHBVEgAAAAAACAAAAAHunqIsJ+NL0BSPxBCOCPdOj1BIsgEAPAsMFO6eoiwn40vQFI/EEI4I906PUEiyEsAfDAR2b3RlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtSOQFCDED0hqysBCKImuYOY/i0/4cTPogx88Ca0aZmWauY1uAU7e4ZzvzbUh7FtAjE8kKPhmf97fNvGaG61fVzMFQ5LujzKAwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CQVbnsyc=]"}
2023-01-11T09:13:05.0509840Z     logger.go:130: 2023-01-11T09:08:00.827Z	DEBUG	done processing headers	{"headerIndex": 6, "blockHeight": 5, "took": "261.991µs"}
2023-01-11T09:13:05.0510720Z     logger.go:130: 2023-01-11T09:08:00.896Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0511780Z     logger.go:130: 2023-01-11T09:08:00.897Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2023-01-11T09:13:05.0513740Z     logger.go:130: 2023-01-11T09:08:00.908Z	DEBUG	processing rpc request	{"method": "invokefunction", "params": "[ef4073a0f2b305a38ec4050e4d3d28bc40ea63f5 getAccountState ]"}
2023-01-11T09:13:05.0515090Z     logger.go:130: 2023-01-11T09:08:00.909Z	DEBUG	processing rpc request	{"method": "invokefunction", "params": "[ef4073a0f2b305a38ec4050e4d3d28bc40ea63f5 decimals ]"}
2023-01-11T09:13:05.0515610Z address was not provided
2023-01-11T09:13:05.0516090Z additional arguments given while this command expects none
2023-01-11T09:13:05.0516970Z     logger.go:130: 2023-01-11T09:08:00.913Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0517970Z     logger.go:130: 2023-01-11T09:08:00.914Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2023-01-11T09:13:05.0518920Z     logger.go:130: 2023-01-11T09:08:00.924Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2023-01-11T09:13:05.0520210Z     logger.go:130: 2023-01-11T09:08:00.925Z	DEBUG	processing rpc request	{"method": "invokescript", "params": "[DCECs2Ir9AF73+MXxYrtX0x1PyBrfbiWBG+n13S7xL9/jcIRwB8ME3VucmVnaXN0ZXJDYW5kaWRhdGUMFPVj6kC8KD1NDgXEjqMFs/Kgc0DvQWJ9W1I5 ]"}
2023-01-11T09:13:05.0520880Z ==================
2023-01-11T09:13:05.0521270Z WARNING: DATA RACE
2023-01-11T09:13:05.0521620Z Write at 0x00c0004c95a0 by goroutine 171:
2023-01-11T09:13:05.0522520Z   github.com/nspcc-dev/neo-go/pkg/core/native.(*NEO).ComputeNextBlockValidators()
2023-01-11T09:13:05.0523600Z       /Users/runner/work/neo-go/neo-go/pkg/core/native/native_neo.go:1075 +0x2a5
2023-01-11T09:13:05.0524420Z   github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).GetValidators()
2023-01-11T09:13:05.0525130Z       /Users/runner/work/neo-go/neo-go/pkg/core/blockchain.go:2572 +0x94
2023-01-11T09:13:05.0525780Z   github.com/nspcc-dev/neo-go/pkg/consensus.(*service).getValidators()
2023-01-11T09:13:05.0526890Z       /Users/runner/work/neo-go/neo-go/pkg/consensus/consensus.go:667 +0x89
2023-01-11T09:13:05.0527830Z   github.com/nspcc-dev/neo-go/pkg/consensus.(*service).getValidators-fm()
2023-01-11T09:13:05.0528590Z       <autogenerated>:1 +0x64
2023-01-11T09:13:05.0529390Z   github.com/nspcc-dev/dbft.(*Context).Fill()
2023-01-11T09:13:05.0530240Z       /Users/runner/go/pkg/mod/github.com/nspcc-dev/dbft@v0.0.0-20221020093431-31c1bbdc74f2/context.go:241 +0x53b
2023-01-11T09:13:05.0530930Z   github.com/nspcc-dev/dbft.(*Context).makePrepareRequest()
2023-01-11T09:13:05.0531760Z       /Users/runner/go/pkg/mod/github.com/nspcc-dev/dbft@v0.0.0-20221020093431-31c1bbdc74f2/send.go:19 +0x30
2023-01-11T09:13:05.0532420Z   github.com/nspcc-dev/dbft.(*DBFT).sendPrepareRequest()
2023-01-11T09:13:05.0533400Z       /Users/runner/go/pkg/mod/github.com/nspcc-dev/dbft@v0.0.0-20221020093431-31c1bbdc74f2/send.go:31 +0x4c
2023-01-11T09:13:05.0534090Z   github.com/nspcc-dev/dbft.(*DBFT).OnTimeout()
2023-01-11T09:13:05.0534990Z       /Users/runner/go/pkg/mod/github.com/nspcc-dev/dbft@v0.0.0-20221020093431-31c1bbdc74f2/dbft.go:190 +0x5d6
2023-01-11T09:13:05.0535660Z   github.com/nspcc-dev/neo-go/pkg/consensus.(*service).eventLoop()
2023-01-11T09:13:05.0536450Z       /Users/runner/work/neo-go/neo-go/pkg/consensus/consensus.go:307 +0x555
2023-01-11T09:13:05.0537040Z   github.com/nspcc-dev/neo-go/pkg/consensus.(*service).Start.func1()
2023-01-11T09:13:05.0537900Z       /Users/runner/work/neo-go/neo-go/pkg/consensus/consensus.go:278 +0x39
2023-01-11T09:13:05.0538190Z 
2023-01-11T09:13:05.0538400Z Previous read at 0x00c0004c95a0 by goroutine 299:
2023-01-11T09:13:05.0539010Z   github.com/nspcc-dev/neo-go/pkg/core/native.copyNeoCache()
2023-01-11T09:13:05.0539820Z       /Users/runner/work/neo-go/neo-go/pkg/core/native/native_neo.go:130 +0x184
2023-01-11T09:13:05.0540450Z   github.com/nspcc-dev/neo-go/pkg/core/native.(*NeoCache).Copy()
2023-01-11T09:13:05.0541310Z       /Users/runner/work/neo-go/neo-go/pkg/core/native/native_neo.go:121 +0x9d
2023-01-11T09:13:05.0542260Z   github.com/nspcc-dev/neo-go/pkg/core/dao.(*Simple).getCache()
2023-01-11T09:13:05.0543240Z       /Users/runner/work/neo-go/neo-go/pkg/core/dao/dao.go:930 +0xf6
2023-01-11T09:13:05.0544130Z   github.com/nspcc-dev/neo-go/pkg/core/dao.(*Simple).GetRWCache()
2023-01-11T09:13:05.0545380Z       /Users/runner/work/neo-go/neo-go/pkg/core/dao/dao.go:912 +0xae
2023-01-11T09:13:05.0546310Z   github.com/nspcc-dev/neo-go/pkg/core/native.(*NEO).UnregisterCandidateInternal()
2023-01-11T09:13:05.0547160Z       /Users/runner/work/neo-go/neo-go/pkg/core/native/native_neo.go:794 +0x14f
2023-01-11T09:13:05.0547960Z   github.com/nspcc-dev/neo-go/pkg/core/native.(*NEO).unregisterCandidate()
2023-01-11T09:13:05.0548780Z       /Users/runner/work/neo-go/neo-go/pkg/core/native/native_neo.go:781 +0x97
2023-01-11T09:13:05.0549460Z   github.com/nspcc-dev/neo-go/pkg/core/native.(*NEO).unregisterCandidate-fm()
2023-01-11T09:13:05.0550030Z       <autogenerated>:1 +0x6d
2023-01-11T09:13:05.0550560Z   github.com/nspcc-dev/neo-go/pkg/core/native.Call()
2023-01-11T09:13:05.0551700Z       /Users/runner/work/neo-go/neo-go/pkg/core/native/interop.go:63 +0x1544
2023-01-11T09:13:05.0552370Z   github.com/nspcc-dev/neo-go/pkg/core/interop.(*Context).SyscallHandler()
2023-01-11T09:13:05.0553140Z       /Users/runner/work/neo-go/neo-go/pkg/core/interop/context.go:323 +0x2ef
2023-01-11T09:13:05.0554080Z   github.com/nspcc-dev/neo-go/pkg/core/interop.(*Context).SyscallHandler-fm()
2023-01-11T09:13:05.0554600Z       <autogenerated>:1 +0x4b
2023-01-11T09:13:05.0555090Z   github.com/nspcc-dev/neo-go/pkg/vm.(*VM).execute()
2023-01-11T09:13:05.0555780Z       /Users/runner/work/neo-go/neo-go/pkg/vm/vm.go:1473 +0x1390
2023-01-11T09:13:05.0556340Z   github.com/nspcc-dev/neo-go/pkg/vm.(*VM).step()
2023-01-11T09:13:05.0557010Z       /Users/runner/work/neo-go/neo-go/pkg/vm/vm.go:472 +0x16f
2023-01-11T09:13:05.0557660Z   github.com/nspcc-dev/neo-go/pkg/vm.(*VM).Run()
2023-01-11T09:13:05.0558710Z       /Users/runner/work/neo-go/neo-go/pkg/vm/vm.go:444 +0x28f
2023-01-11T09:13:05.0559530Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).runScriptInVM()
2023-01-11T09:13:05.0560390Z       /Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:2119 +0x105
2023-01-11T09:13:05.0561080Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).invokescript()
2023-01-11T09:13:05.0561850Z       /Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:1931 +0xee
2023-01-11T09:13:05.0562500Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).handleIn()
2023-01-11T09:13:05.0563290Z       /Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:540 +0x927
2023-01-11T09:13:05.0563870Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).handleRequest()
2023-01-11T09:13:05.0564660Z       /Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:511 +0x190
2023-01-11T09:13:05.0565320Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).handleHTTPRequest()
2023-01-11T09:13:05.0566110Z       /Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:504 +0x1317
2023-01-11T09:13:05.0566840Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).handleHTTPRequest-fm()
2023-01-11T09:13:05.0567320Z       <autogenerated>:1 +0x57
2023-01-11T09:13:05.0567710Z   net/http.HandlerFunc.ServeHTTP()
2023-01-11T09:13:05.0568390Z       /Users/runner/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:2109 +0x4d
2023-01-11T09:13:05.0568970Z   net/http.serverHandler.ServeHTTP()
2023-01-11T09:13:05.0613030Z       /Users/runner/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:2947 +0x641
2023-01-11T09:13:05.0613370Z   net/http.(*conn).serve()
2023-01-11T09:13:05.0613820Z       /Users/runner/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:1991 +0xbe4
2023-01-11T09:13:05.0614140Z   net/http.(*Server).Serve.func3()
2023-01-11T09:13:05.0614600Z       /Users/runner/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:3102 +0x58
2023-01-11T09:13:05.0614810Z 
2023-01-11T09:13:05.0614930Z Goroutine 171 (running) created at:
2023-01-11T09:13:05.0615550Z   github.com/nspcc-dev/neo-go/pkg/consensus.(*service).Start()
2023-01-11T09:13:05.0616180Z       /Users/runner/work/neo-go/neo-go/pkg/consensus/consensus.go:278 +0x2cb
2023-01-11T09:13:05.0616700Z   github.com/nspcc-dev/neo-go/pkg/network.(*Server).tryStartServices()
2023-01-11T09:13:05.0617300Z       /Users/runner/work/neo-go/neo-go/pkg/network/server.go:545 +0x221
2023-01-11T09:13:05.0617810Z   github.com/nspcc-dev/neo-go/pkg/network.(*Server).Start()
2023-01-11T09:13:05.0618400Z       /Users/runner/work/neo-go/neo-go/pkg/network/server.go:272 +0x298
2023-01-11T09:13:05.0618890Z   github.com/nspcc-dev/neo-go/internal/testcli.NewTestChain.func2()
2023-01-11T09:13:05.0619510Z       /Users/runner/work/neo-go/neo-go/internal/testcli/executor.go:166 +0x47
2023-01-11T09:13:05.0619710Z 
2023-01-11T09:13:05.0619820Z Goroutine 299 (running) created at:
2023-01-11T09:13:05.0620060Z   net/http.(*Server).Serve()
2023-01-11T09:13:05.0620520Z       /Users/runner/hostedtoolcache/go/1.19.4/x64/src/net/http/server.go:3102 +0x837
2023-01-11T09:13:05.0621040Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).Start.func1()
2023-01-11T09:13:05.0621670Z       /Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:358 +0x78
2023-01-11T09:13:05.0622190Z   github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.(*Server).Start.func3()
2023-01-11T09:13:05.0622810Z       /Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/server.go:363 +0x47
2023-01-11T09:13:05.0623080Z ==================
2023-01-11T09:13:05.0623760Z     logger.go:130: 2023-01-11T09:08:00.930Z	DEBUG	done processing headers	{"headerIndex": 7, "blockHeight": 6, "took": "235.76µs"}
2023-01-11T09:13:05.0624490Z     logger.go:130: 2023-01-11T09:08:00.933Z	DEBUG	processing rpc request	{"method": "getblockcount", "params": "[]"}
2023-01-11T09:13:05.0626660Z     logger.go:130: 2023-01-11T09:08:00.934Z	DEBUG	processing rpc request	{"method": "calculatenetworkfee", "params": "[AAILtWxI8y0AAAAAAAAAAAAAAAAACQAAAAHunqIsJ+NL0BSPxBCOCPdOj1BIsgEAVwwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CEcAfDBN1bnJlZ2lzdGVyQ2FuZGlkYXRlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtSOQEAKAwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CQVbnsyc=]"}
2023-01-11T09:13:05.0629530Z     logger.go:130: 2023-01-11T09:08:00.936Z	DEBUG	processing rpc request	{"method": "sendrawtransaction", "params": "[AAILtWxI8y0AAAAAAOi+EgAAAAAACQAAAAHunqIsJ+NL0BSPxBCOCPdOj1BIsgEAVwwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CEcAfDBN1bnJlZ2lzdGVyQ2FuZGlkYXRlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtSOQFCDEAiBEnGvKU+j2LuXhSSoNm/HzuuP5TK5PcWcoM0OSVCPX3EDCygTs/ihsvwx2oMMwxEYbFtHqppvhAh1KVEIg6eKAwhArNiK/QBe9/jF8WK7V9MdT8ga324lgRvp9d0u8S/f43CQVbnsyc=]"}
2023-01-11T09:13:05.0630970Z     logger.go:130: 2023-01-11T09:08:01.036Z	DEBUG	done processing headers	{"headerIndex": 8, "blockHeight": 7, "took": "222.728µs"}
2023-01-11T09:13:05.0631290Z No address specified
2023-01-11T09:13:05.0631530Z this command only accepts one address
2023-01-11T09:13:05.0631840Z additional arguments given while this command expects none
2023-01-11T09:13:05.0632160Z additional arguments given while this command expects none
2023-01-11T09:13:05.0632810Z     logger.go:130: 2023-01-11T09:08:01.140Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:49284"}
2023-01-11T09:13:05.0633550Z     logger.go:130: 2023-01-11T09:08:01.140Z	DEBUG	done processing headers	{"headerIndex": 9, "blockHeight": 8, "took": "336.469µs"}
2023-01-11T09:13:05.0634350Z     logger.go:130: 2023-01-11T09:08:01.141Z	INFO	persisted to disk	{"blocks": 9, "keys": 260, "headerHeight": 9, "blockHeight": 9, "took": "234.735µs"}
2023-01-11T09:13:05.0634820Z     testing.go:1319: race detected during execution of test
2023-01-11T09:13:05.0635210Z --- FAIL: TestRegisterCandidate (0.85s)
```

**Comment by AnnaShaleva** [comment]
https://github.com/nspcc-dev/neo-go/pull/3110 may be closely related to this issue. We need to ensure that this bug still exists after https://github.com/nspcc-dev/neo-go/pull/3110 fix (@AliceInHunterland, you may search through our failing test logs in the https://github.com/nspcc-dev/neo-go/actions/workflows/tests.yml).

**Comment by AliceInHunterland** [comment]
@AnnaShaleva it seems that logs of jobs before #3110 are unavailable (example https://github.com/nspcc-dev/neo-go/actions/runs/6459577009/job/17535640232 ), locally with git checkout 96449d803a91b258e42bed8a8e1fb8970cbe3383^  TestRegisterCandidate is non-reproducible and here too (https://github.com/nspcc-dev/neo-go/actions/runs/7797187985/job/21263311972?pr=3304).

**Comment by roman-khimov** [comment]
None of these issues are reproduced reliably, so we can fix things we can and know how to fix and then see if any of the rest reproduce in like a month.

**Comment by AnnaShaleva** [comment]
> logs of jobs before https://github.com/nspcc-dev/neo-go/pull/3110 are unavailable

Check out several recently failing tests those logs are still available at https://github.com/nspcc-dev/neo-go/actions/workflows/tests.yml. Any failing job is suit, because we haven't change the logic of this test for quite a long time. If this issue is not reproduced, then I suppose that the problem was solved by native cache refactoring in #3110.

**Comment by AnnaShaleva** [comment]
OK, after discussion the result is the following: latest failing testing jobs do not contain failing TestRegisterCandidate test. I think, that this problem was solved by Neo cache refactoring and I'd suggest to close this issue as completed if no TestRegisterCandidate failures discovered in the nearest future.
